extensions/nanoarrow_ipc/src/nanoarrow/nanoarrow_ipc.h

```diff
@@ -80,8 +129,33 @@ ArrowErrorCode ArrowIpcReaderDecode(struct ArrowIpcReader* reader,
                                     struct ArrowBufferView data,
                                     struct ArrowError* error);
 
-#endif
+ArrowErrorCode ArrowIpcReaderGetSchema(struct ArrowIpcReader* reader,
+                                       struct ArrowSchema* out, struct ArrowError* error);
+
+ArrowErrorCode ArrowIpcReaderGetArray(struct ArrowIpcReader* reader,
+                                      struct ArrowBufferView body, int64_t i,
+                                      struct ArrowArray* out, struct ArrowError* error);
+
+ArrowErrorCode ArrowIpcReaderSetSchema(struct ArrowIpcReader* reader,
+                                       struct ArrowSchema* schema,
+                                       struct ArrowError* error);
+
+struct ArrowIpcField {
+  struct ArrowArrayView* array_view;
+  int64_t buffer_offset;
+};
+
+struct ArrowIpcReaderPrivate {
```

Comment:
nit: maybe indicate that it's only here for testing?

(alternatively, you could keep it in the .c file and copy-paste it to the .cc file in an 'extern C' block)